checker/src/org/checkerframework/checker/genericeffects/GenericEffectVisitor.java

```diff
@@ -22,32 +18,55 @@
 
     protected final boolean debugSpew;
     private GenericEffectLattice genericEffect;
-
+    private GenericEffectExtension extension;
     // effStack and currentMethods should always be the same size.
     protected final Stack<Class<? extends Annotation>> effStack;
     protected final Stack<MethodTree> currentMethods;
 
-    public GenericEffectVisitor(BaseTypeChecker checker) {
+    //fields for variables
+    protected final Stack<Class<? extends Annotation>> varEffStack;
+    protected final Stack<VariableTree> currentVars;
+
+    /**
+     * Constructor that takes passes the checker to the superclass and takes in a GenericEffectExtension object.
+     * The effect stack for methods and variables are set up in the constructor.
+     *
+     * @param checker The checker that allows the Casting Effects Checker to function.
+     * @param ext An GenericEffectExtension object that provides the developer with more functions dealing with specific tree nodes.
+     */
+    public GenericEffectVisitor(BaseTypeChecker checker, GenericEffectExtension ext) {
         super(checker);
         assert (checker instanceof GenericEffectChecker);
         debugSpew = checker.getLintOption("debugSpew", false);
 
         effStack = new Stack<Class<? extends Annotation>>();
         currentMethods = new Stack<MethodTree>();
 
+        //added this assignment
+        varEffStack = new Stack<Class<? extends Annotation>>();;
+        currentVars = new Stack<VariableTree>();
+        extension = ext;
+
         genericEffect = ((GenericEffectChecker) checker).getEffectLattice();
     }
 
-    // Method to instantiate the factory class for the checker
+    /**
+     * Method to instantiate the factory class for the checker.
+     *
+     * @return The type factory of the checker.
+     */
     @Override
     protected GenericEffectTypeFactory createTypeFactory() {
         return new GenericEffectTypeFactory(checker, debugSpew);
     }
 
+    //TODO: Determine what the use of this method is.
     @Override
     protected void checkMethodInvocability(
-            AnnotatedExecutableType method, MethodInvocationTree node) {}
+            AnnotatedExecutableType method, MethodInvocationTree node) {
+    }
 
+    //TODO: Determine what the use of this method is.
```

Comment:
There's a comment about this in the GuiEffectChecker, whose code forms the basis for this class.  Essentially, we perform our own override checks.  Same for ```checkMethodInvocability``` above.  You should document them, following the explanations given in the GuiEffectChecker code.